code/src/cljs/sixsq/nuvla/ui/edges/events.cljs

```diff
@@ -124,10 +124,19 @@
                                      status (str " (" status ")"))
                     :content message
                     :type    :error})])
-      {:db (assoc db ::spec/next-heartbeats-offline-edges (zipmap
-                                                            (map :parent resources)
-                                                            (map :next-heartbeat resources))
-                     ::main-spec/loading? false)})))
+      {:db (assoc db ::spec/next-heartbeats-offline-edges (as-> (filter (fn [edge] (= (:online edge) false)) resources) offline-edges
+                                                            (zipmap
+                                                             (map :parent offline-edges)
+                                                             (map :next-heartbeat offline-edges)))
+                  ::main-spec/loading? false)
+       :fx [[:dispatch [::set-nuvlaboxes-engine-versions resources]]]})))
+
+(reg-event-fx
```

Comment:
why another event, it can be an assoc in the db within ::set-nuvlaboxes-next-heartbeats event. By the way the ::get-nuvlaboxes-next-heartbeats and ::set-nuvlaboxes-next-heartbeats should be renamed in a generic way because they are doing more than that now

Reply:
I did it that way first, doing everything inside `::set-nuvlaboxes-next-heartbeats` event but then I though new event would be better, because smaller and theoretically reusable from somewhere else. But then I think I should have renamed `::get-nuvlaboxes-next-heartbeats` to something more generic and returned the `:dispatch` for the event for setting the engine versions from there and definitely not from the `::set-nuvlaboxes-next-heartbeats`.

But I think you're right and this way doesn't make sense anyway: I make it into one event and also put the version and next-heartbeat into the same map, not two maps for next-heartbeat and version separately.